packages/html-manager/src/libembed.ts

```diff
@@ -57,47 +59,44 @@ export function renderWidgets(
  * Additionally, if the script tag has a prior img sibling with class
  * 'jupyter-widget', then that img tag is deleted.
  */
-function renderManager(
+async function renderManager(
   element: HTMLElement,
   widgetState: any,
   managerFactory: () => HTMLManager
-): void {
+): Promise<void> {
   const valid = model_validate(widgetState);
   if (!valid) {
-    console.error('Model state has errors.', model_validate.errors);
+    throw new Error(`Model state has errors: ${model_validate.errors}`);
   }
   const manager = managerFactory();
-  manager.set_state(widgetState).then(function(models) {
-    const tags = element.querySelectorAll(
-      'script[type="application/vnd.jupyter.widget-view+json"]'
-    );
-    for (let i = 0; i != tags.length; ++i) {
-      const viewtag = tags[i];
+  const models = await manager.set_state(widgetState);
+  const tags = element.querySelectorAll(
+    'script[type="application/vnd.jupyter.widget-view+json"]'
+  );
+  await Promise.all(
+    Array.from(tags).map(async viewtag => {
       const widgetViewObject = JSON.parse(viewtag.innerHTML);
       const valid = view_validate(widgetViewObject);
       if (!valid) {
-        console.error('View state has errors.', view_validate.errors);
+        throw new Error(`View state has errors: ${view_validate.errors}`);
```

Comment:
Note that `Promise.all` has a fail-fast behavior (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all), so if one view has an issue, this change will cause *all* views to not be displayed.

Reply:
Ah, actually, the fail-fast will just cause the outer promise to reject, it will still keep processing the other promises. My bad :)

Reply:
So we're all good, then?

Reply:
ping @vidartf

Reply:
ah, I didn't see that you marked this conversation as "resolved", which indicates we are all good on this after all.